docs/maybe.md

```diff
@@ -0,0 +1,52 @@
+###E247.Fun
+#Maybe&lt;T&gt;
+
+```Maybe<T>``` is used to represent a value that might not be available. While it is an option to use ```null``` for reference types, it isn't explicit in the type system and returning ```null``` can lead to problems when it's not handled properly, using ```Maybe<T>``` instead makes it obvious that the potential lack of a value needs to be handled, and provides some handy methods for dealing with boilerplate based around that.
+
+##Making a Maybe
+To convert a typed value to a ```Maybe``` you can simply call the ```.ToMaybe()``` extension method, this will even handle if that value is ```null``` by creating an empty ```Maybe``` of that type.
+
+##Checking for a Value
+Before using the value in a ```Maybe``` you should first check if it actually contains one. This can be done by testing the boolean value of the ```.HasValue``` property or the ```.Any()``` method. 
+
+It's also possible to match on both potential states of a ```Maybe``` using the ```.Match``` method which takes 2 functions, one for handling each state:
+
+    someMaybeValue.Match(
+        Some: theValue => Console.WriteLine($"It was {theValue}"),
+        None: () => Console.WriteLine("It was empty"));
+
```

Comment:
I have doubts about presenting as a first example a use of .Match just for side effect. Probably we should stress that .Match is an expression that returns a value of a specific type, and that it should be used lske that whenever possible.
Maybe it would be better in this case to avoid having the Console.WriteLine, and just show returning a string. Of course we can then add that .Match can be used also for a side effect, but I would add that it should be avoided if it's reasonable to do so..